logicrunner/goplugin/preprocessor/main.go

```diff
@@ -174,11 +177,47 @@ func generateContractWrapper(fileName string, out io.Writer) error {
 		panic("Contract must be in main package")
 	}
 
-	code := "package " + packageName + "\n\n"
-	code += generateWrappers(parsed) + "\n"
-	code += generateExports(parsed) + "\n"
+	var methods []map[string]interface{}
+	for _, method := range parsed.methods[parsed.contract] {
+		argsInit, argsList := generateZeroListOfTypes(parsed, "args", method.Type.Params)
 
-	_, err = out.Write([]byte(code))
+		rets := []string{}
+		for i := range method.Type.Results.List {
+			rets = append(rets, fmt.Sprintf("ret%d", i))
+		}
+		resultList := strings.Join(rets, ", ")
+
+		info := map[string]interface{}{
+			"Name":              method.Name.Name,
+			"ArgumentsZeroList": argsInit,
+			"Results":           resultList,
+			"Arguments":         argsList,
+		}
+		methods = append(methods, info)
+	}
+
+	_, currentFile, _, ok := runtime.Caller(0)
```

Comment:
Maybe GetTemplateDir() ?

Reply:
I dont think its gonna make this 4 lines more clear